src/coreclr/jit/llvm.cpp

```diff
@@ -1643,47 +1405,298 @@ llvm::DILocation* createDebugFunctionAndDiLocation(struct DebugMetadata debugMet
+                    {
+                        callUse.ReplaceWith(_compiler, indirNode);
+                    }
+                    else
+                    {
+                        callNode->ClearUnusedValue();
+                    }
+
+                    callNode->gtReturnType = var_types::TYP_VOID;
+                    callNode->ChangeType(TYP_VOID);
+
+                    CurrentRange().InsertBefore(callNode, shadowStackVar);
+                    CurrentRange().InsertAfter(shadowStackVar, offset);
+                    CurrentRange().InsertAfter(offset, returnValueAddress);
+                    CurrentRange().InsertAfter(returnValueAddress, addrStore);
+                    CurrentRange().InsertAfter(addrStore, returnAddrLcl);
+                    CurrentRange().InsertAfter(callNode, returnAddrLcl2);
+                    CurrentRange().InsertAfter(returnAddrLcl2, indirNode);
+                }
+            }
+            else if (node->OperIs(GT_RETURN) && _functionSigHasReturnAddress)
+            {
+                var_types originalReturnType = node->TypeGet();
+                if(node->TypeIs(TYP_VOID))
+                {
+                    /* TODO-LLVM: retbuf .   compHasRetBuffArg doesn't seem to have an implementation */
+                    failFunctionCompilation();
+                }
+                //TODO-LLVM: how to create a GT_STORE_OBJ node?
+                if (originalReturnType == TYP_STRUCT)
+                {
+                    failFunctionCompilation();
+                }
+
+                assert(_retAddressLclNum == -1);
+                _retAddressLclNum = _compiler->lvaGrabTemp(true DEBUGARG("shadowstack"));
+                LclVarDsc* retAddressVarDsc = _compiler->lvaGetDesc(_retAddressLclNum);
+                retAddressVarDsc->lvIsParam = 1;
+                retAddressVarDsc->lvType = var_types::TYP_I_IMPL;
+
+                GenTreeLclVar* retAddressLocal = _compiler->gtNewLclvNode(_retAddressLclNum, TYP_I_IMPL);
+
+                GenTree* storeNode = _compiler->gtNewOperNode(originalReturnType == TYP_STRUCT ? GT_STORE_OBJ : GT_STOREIND, originalReturnType, retAddressLocal, node->AsOp()->gtOp1);
+                storeNode->gtFlags |= GTF_IND_TGT_NOT_HEAP; // No RhpAssignRef required
+
+                GenTreeOp* retNode = node->AsOp();
+                retNode->gtOp1 = nullptr;
+                node->ChangeType(TYP_VOID);
+
+                CurrentRange().InsertBefore(node, retAddressLocal);
+                CurrentRange().InsertAfter(retAddressLocal, storeNode);
+            }
+        }
     }
 }
 
+//------------------------------------------------------------------------
+// Convert GT_STORE_LCL_VAR and GT_LCL_VAR to use the shadow stack when the local needs to be GC tracked
+//
+void Llvm::PlaceAndConvertShadowStackLocals()
+{
+    _shadowStackLocalsSize = 0;
+    if (_compiler->lvaCount == 0)
+    {
+        return;
+    }
+
+    std::vector<LclVarDsc*> locals;
```

Comment:
```suggestion
    std::vector<LclVarDsc*> locals();
```
Initializing the `locals` local...